### ISSUE TITLE
[0/1] Re-running CMake Loop

### ISSUE THREAD

**Issue by dtischler** [edgeimpulse/example-standalone-inferencing-espressif-esp32#5]
After cloning this repo and then adding the folders taken from a normal Studio download `.zip` file as indicated in the Readme, I run a `idf.py build` and end up with this loop:

```
-- Configuring done
-- Generating done
-- Build files have been written to: /home/david/example-standalone-inferencing-espressif-esp32/build
[0/1] Re-running CMake...
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/david/example-standalone-inferencing-espressif-esp32/sdkconfig
-- App "ei_standalone_esp32" version: bc7dfa8
-- Adding linker script /home/david/example-standalone-inferencing-espressif-esp32/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /home/david/esp-idf/components/esp_system/ld/esp32/sections.ld.in
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.api.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-data.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.syscalls.ld
-- Adding linker script /home/david/esp-idf/components/soc/esp32/ld/esp32.peripherals.ld
-- Components: app_trace app_update asio bootloader bootloader_support bt cbor cmock coap console cxx driver efuse esp-tls esp32 esp_adc_cal esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_ipc esp_lcd esp_local_ctrl esp_netif esp_phy esp_pm esp_ringbuf esp_rom esp_serial_slave_link esp_system esp_timer esp_websocket_client esp_wifi espcoredump esptool_py expat fatfs freemodbus freertos hal heap idf_test ieee802154 jsmn json libsodium log lwip main mbedtls mdns mqtt newlib nghttp nvs_flash openssl openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs tcp_transport tcpip_adapter tinyusb ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: /home/david/esp-idf/components/app_trace /home/david/esp-idf/components/app_update /home/david/esp-idf/components/asio /home/david/esp-idf/components/bootloader /home/david/esp-idf/components/bootloader_support /home/david/esp-idf/components/bt /home/david/esp-idf/components/cbor /home/david/esp-idf/components/cmock /home/david/esp-idf/components/coap /home/david/esp-idf/components/console /home/david/esp-idf/components/cxx /home/david/esp-idf/components/driver /home/david/esp-idf/components/efuse /home/david/esp-idf/components/esp-tls /home/david/esp-idf/components/esp32 /home/david/esp-idf/components/esp_adc_cal /home/david/esp-idf/components/esp_common /home/david/esp-idf/components/esp_eth /home/david/esp-idf/components/esp_event /home/david/esp-idf/components/esp_gdbstub /home/david/esp-idf/components/esp_hid /home/david/esp-idf/components/esp_http_client /home/david/esp-idf/components/esp_http_server /home/david/esp-idf/components/esp_https_ota /home/david/esp-idf/components/esp_https_server /home/david/esp-idf/components/esp_hw_support /home/david/esp-idf/components/esp_ipc /home/david/esp-idf/components/esp_lcd /home/david/esp-idf/components/esp_local_ctrl /home/david/esp-idf/components/esp_netif /home/david/esp-idf/components/esp_phy /home/david/esp-idf/components/esp_pm /home/david/esp-idf/components/esp_ringbuf /home/david/esp-idf/components/esp_rom /home/david/esp-idf/components/esp_serial_slave_link /home/david/esp-idf/components/esp_system /home/david/esp-idf/components/esp_timer /home/david/esp-idf/components/esp_websocket_client /home/david/esp-idf/components/esp_wifi /home/david/esp-idf/components/espcoredump /home/david/esp-idf/components/esptool_py /home/david/esp-idf/components/expat /home/david/esp-idf/components/fatfs /home/david/esp-idf/components/freemodbus /home/david/esp-idf/components/freertos /home/david/esp-idf/components/hal /home/david/esp-idf/components/heap /home/david/esp-idf/components/idf_test /home/david/esp-idf/components/ieee802154 /home/david/esp-idf/components/jsmn /home/david/esp-idf/components/json /home/david/esp-idf/components/libsodium /home/david/esp-idf/components/log /home/david/esp-idf/components/lwip /home/david/example-standalone-inferencing-espressif-esp32/main /home/david/esp-idf/components/mbedtls /home/david/esp-idf/components/mdns /home/david/esp-idf/components/mqtt /home/david/esp-idf/components/newlib /home/david/esp-idf/components/nghttp /home/david/esp-idf/components/nvs_flash /home/david/esp-idf/components/openssl /home/david/esp-idf/components/openthread /home/david/esp-idf/components/partition_table /home/david/esp-idf/components/perfmon /home/david/esp-idf/components/protobuf-c /home/david/esp-idf/components/protocomm /home/david/esp-idf/components/pthread /home/david/esp-idf/components/sdmmc /home/david/esp-idf/components/soc /home/david/esp-idf/components/spi_flash /home/david/esp-idf/components/spiffs /home/david/esp-idf/components/tcp_transport /home/david/esp-idf/components/tcpip_adapter /home/david/esp-idf/components/tinyusb /home/david/esp-idf/components/ulp /home/david/esp-idf/components/unity /home/david/esp-idf/components/usb /home/david/esp-idf/components/vfs /home/david/esp-idf/components/wear_levelling /home/david/esp-idf/components/wifi_provisioning /home/david/esp-idf/components/wpa_supplicant /home/david/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/david/example-standalone-inferencing-espressif-esp32/build
[0/1] Re-running CMake...
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/david/example-standalone-inferencing-espressif-esp32/sdkconfig
-- App "ei_standalone_esp32" version: bc7dfa8
-- Adding linker script /home/david/example-standalone-inferencing-espressif-esp32/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /home/david/esp-idf/components/esp_system/ld/esp32/sections.ld.in
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.api.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.libgcc.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-data.ld
-- Adding linker script /home/david/esp-idf/components/esp_rom/esp32/ld/esp32.rom.syscalls.ld
-- Adding linker script /home/david/esp-idf/components/soc/esp32/ld/esp32.peripherals.ld
-- Components: app_trace app_update asio bootloader bootloader_support bt cbor cmock coap console cxx driver efuse esp-tls esp32 esp_adc_cal esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_ipc esp_lcd esp_local_ctrl esp_netif esp_phy esp_pm esp_ringbuf esp_rom esp_serial_slave_link esp_system esp_timer esp_websocket_client esp_wifi espcoredump esptool_py expat fatfs freemodbus freertos hal heap idf_test ieee802154 jsmn json libsodium log lwip main mbedtls mdns mqtt newlib nghttp nvs_flash openssl openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs tcp_transport tcpip_adapter tinyusb ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: /home/david/esp-idf/components/app_trace /home/david/esp-idf/components/app_update /home/david/esp-idf/components/asio /home/david/esp-idf/components/bootloader /home/david/esp-idf/components/bootloader_support /home/david/esp-idf/components/bt /home/david/esp-idf/components/cbor /home/david/esp-idf/components/cmock /home/david/esp-idf/components/coap /home/david/esp-idf/components/console /home/david/esp-idf/components/cxx /home/david/esp-idf/components/driver /home/david/esp-idf/components/efuse /home/david/esp-idf/components/esp-tls /home/david/esp-idf/components/esp32 /home/david/esp-idf/components/esp_adc_cal /home/david/esp-idf/components/esp_common /home/david/esp-idf/components/esp_eth /home/david/esp-idf/components/esp_event /home/david/esp-idf/components/esp_gdbstub /home/david/esp-idf/components/esp_hid /home/david/esp-idf/components/esp_http_client /home/david/esp-idf/components/esp_http_server /home/david/esp-idf/components/esp_https_ota /home/david/esp-idf/components/esp_https_server /home/david/esp-idf/components/esp_hw_support /home/david/esp-idf/components/esp_ipc /home/david/esp-idf/components/esp_lcd /home/david/esp-idf/components/esp_local_ctrl /home/david/esp-idf/components/esp_netif /home/david/esp-idf/components/esp_phy /home/david/esp-idf/components/esp_pm /home/david/esp-idf/components/esp_ringbuf /home/david/esp-idf/components/esp_rom /home/david/esp-idf/components/esp_serial_slave_link /home/david/esp-idf/components/esp_system /home/david/esp-idf/components/esp_timer /home/david/esp-idf/components/esp_websocket_client /home/david/esp-idf/components/esp_wifi /home/david/esp-idf/components/espcoredump /home/david/esp-idf/components/esptool_py /home/david/esp-idf/components/expat /home/david/esp-idf/components/fatfs /home/david/esp-idf/components/freemodbus /home/david/esp-idf/components/freertos /home/david/esp-idf/components/hal /home/david/esp-idf/components/heap /home/david/esp-idf/components/idf_test /home/david/esp-idf/components/ieee802154 /home/david/esp-idf/components/jsmn /home/david/esp-idf/components/json /home/david/esp-idf/components/libsodium /home/david/esp-idf/components/log /home/david/esp-idf/components/lwip /home/david/example-standalone-inferencing-espressif-esp32/main /home/david/esp-idf/components/mbedtls /home/david/esp-idf/components/mdns /home/david/esp-idf/components/mqtt /home/david/esp-idf/components/newlib /home/david/esp-idf/components/nghttp /home/david/esp-idf/components/nvs_flash /home/david/esp-idf/components/openssl /home/david/esp-idf/components/openthread /home/david/esp-idf/components/partition_table /home/david/esp-idf/components/perfmon /home/david/esp-idf/components/protobuf-c /home/david/esp-idf/components/protocomm /home/david/esp-idf/components/pthread /home/david/esp-idf/components/sdmmc /home/david/esp-idf/components/soc /home/david/esp-idf/components/spi_flash /home/david/esp-idf/components/spiffs /home/david/esp-idf/components/tcp_transport /home/david/esp-idf/components/tcpip_adapter /home/david/esp-idf/components/tinyusb /home/david/esp-idf/components/ulp /home/david/esp-idf/components/unity /home/david/esp-idf/components/usb /home/david/esp-idf/components/vfs /home/david/esp-idf/components/wear_levelling /home/david/esp-idf/components/wifi_provisioning /home/david/esp-idf/components/wpa_supplicant /home/david/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/david/example-standalone-inferencing-espressif-esp32/build
[0/1] Re-running CMake...
-- Building ESP-IDF components for target esp32
-- Project sdkconfig file /home/david/example-standalone-inferencing-espressif-esp32/sdkconfig
-- App "ei_standalone_esp32" version: bc7dfa8
-- Adding linker script /home/david/example-standalone-inferencing-espressif-esp32/build/esp-idf/esp_system/ld/memory.ld
```

Eventually it will end with this:

```
avid/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/david/example-standalone-inferencing-espressif-esp32/build
ninja: error: manifest 'build.ninja' still dirty after 100 tries

ninja failed with exit code 1
```

Duplicated on both x86 and Aarch64, Ubuntu 20.04, and ESP developer tools freshly installed via their documentation / instructions.  Are you able to duplicate this behavior @AIWintermuteAI , or have any ideas?

**Comment by AIWintermuteAI** [comment]
Can you try ESP IDF 4.4, as specified in the README? 
https://github.com/edgeimpulse/example-standalone-inferencing-espressif-esp32#software
And before you do that, delete the build folder to make sure you're starting a clean build.

**Comment by dtischler** [comment]
Strange...I did indeed have ESP IDF 4.4 already as I had followed the Getting Started Guide as mentioned in the Readme.  But, for the sake of it, I nuked the folder, re-installed it again, re-ran the build and now it works.  (I actually end up with a `The size of your 'features' array is not correct. Expected 102400 items, but had 0` error, but that is unrelated and I'll troubleshoot that separately. )
